### DISCUSSION TITLE
How do I use this?

### DISCUSSION THREAD

**Discussion by Darin755** [jeremyckahn/chitchatter#154] [General]
I am looking for a replacement for Jami but this seems very complicated. 

I can get two devices to connect to a room but it doesn't work with any browsers with mull browser. This software seems to lack any notification of incompatibility so its possible the browser is blocking something.

Once I got my devices in the same room I was unable to figure out how to get video and audio working. I have no idea if there is a button I need to push but it isn't intuitive. Am I missing something?

**Comment by jeremyckahn** [comment]
Hi @Darin755, I haven’t heard of the Mull browser before this so I haven’t done any testing with it. It’s possible that Chitchatter may not support it. I recommend using Firefox or another more mainstream browser for the best results. Informative connection error messaging isn’t always possible due to the decentralized nature of Chitchatter, but I’m open to ideas for improvements.

Once you’re in a room, you can start sharing audio and video by clicking the green buttons at the top of the screen. Let me know if that works for you.

**Reply by Darin755** [reply]
It didn't but I can try again later. I think mull wasn't working because it blocks webrtc.

**Reply by jeremyckahn** [reply]
I see. Chitchatter requires WebRTC to function, so it sounds like it won’t be compatible with Mull.

**Reply by Darin755** [reply]
Would it be possible to add a notification popup saying that? It would be nice to know what went wrong

**Reply by jeremyckahn** [reply]
Yes, that seems reasonable. Assuming Mull is anything like the Tor browser, it should just be a matter of checking the value of `typeof window.RTCDataChannel`. I'll open an issue to get that implemented.

**Reply by jeremyckahn** [reply]
Filed at #155.

**Reply by Darin755** [reply]
That was fast. Thank you

**Reply by jeremyckahn** [reply]
@Darin755 I've got #161 up for this. Could you please check the build preview URL (https://chitchatter-git-feature-155handle-unsupporte-b5edd3-jeremyckahn.vercel.app/) and let me know if you see an error message about WebRTC being unavailable?

**Reply by Darin755** [reply]
It just searchs for servers indefinitely for me

**Reply by jeremyckahn** [reply]
If you're not seeing a message about WebRTC support, then it's possible your browser is preventing you from connecting to a WebTorrent server (which is necessary for Chitchatter to work). Can you try waiting at least 30 seconds and seeing if the server loading indicator ever changes? It shouldn't search indefinitely. It should eventually indicate a connection error.

**Reply by Darin755** [reply]
I forgot to mention that Mull browser is just a soft fork of Fenix that is hardened for security and privacy. You might want to consider adding a in browser check to make sure that none of the required features are turned blocked. This is especially true if you expect this to be used by people who care more about privacy and security.

I might actually look into opening a PR adds those checks when I get time.

**Reply by Darin755** [reply]
I found a library for detecting webRTC
https://github.com/muaz-khan/DetectRTC

**Reply by jeremyckahn** [reply]
> You might want to consider adding a in browser check to make sure that none of the required features are turned blocked.

This was added in #161. It's possible that there are other missing features in Mull, but I don't have an Android device handy to test and debug with. Were you able to see Chitchatter eventually make a server connection (as shown by the red/yellow/green indicator in the top of the right-hand drawer)? This is what a good connection looks like:

![Chitchatter server connection screenshot](https://github.com/jeremyckahn/chitchatter/assets/366330/f470e743-d692-407e-9547-b38018e591c7)
